### PR TITLE
Add robot-simulator exercise

### PR DESCRIPTION
**Pull request by kytrinyx**

### CODE REVIEW

File: robot-simulator/example.erl [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+             Pid ! {position, Position},
+             loop( Direction, Position )
+      end.
+
+new_direction( north, right ) -> east;
+new_direction( south, right ) -> west;
+new_direction( east, right ) -> south;
+new_direction( west, right ) -> north;
+new_direction( north, left ) -> west;
+new_direction( south, left ) ->	east;
```
Comment (tmcgilchrist) [inline robot-simulator/example.erl]: spacing on `east` super minor /nitpick

File: robot-simulator/robot_simulator_tests.erl [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+	       Robot = robot_simulator:create(),
+	       ?assert( robot_simulator:direction(Robot) =:= undefined ),
+	       ?assert(	robot_simulator:position(Robot) =:= {undefined, undefined} ).
+
+place_test() ->
+            Robot = robot_simulator:create(),
+	    Position = {1, 2},
+	    robot_simulator:place( Robot, north, Position ),
+	    ?assert( robot_simulator:direction(Robot) =:= north ),
+	    ?assert( robot_simulator:position(Robot) =:= Position ).
```
Comment (tmcgilchrist) [inline robot-simulator/robot_simulator_tests.erl]: And if you can be bothered, don't feel obliged, @kytrinyx lining up the tests would be nice too. /nitpick

**Comment by tmcgilchrist** [comment]
Other than me being a pedant, it looks good.

**Comment by kytrinyx** [comment]
There was a mix of tabs and spaces, but I'm not convinced that my editor knows how to erlang. I'll give it a try, and if it doesn't work I'll find a syntax file tomorrow to fix it. Thanks :)

**Comment by kytrinyx** [comment]
Ok, I've given it a shot. Let me know if that looks right to you!

**Comment by tmcgilchrist** [comment]
Much better, thanks :rocket:

**Pull request merged**